### DISCUSSION TITLE
Custom events creation in Self-Hosted

### DISCUSSION THREAD

**Discussion by prk2331** [plausible/analytics#2864] [Self-Hosted Support]
Hello team
Q1. Can we create custom events like add-to-cart, purchase, checkout, etc. through the API?

By events? I am assuming plausible is providing some kind of schema where we can define what events we want to capture, like add-to-cart, purchase, checkout, etc. For example, in the add-to-cart schema,  we  will define what attributes we want to collect from the client side, If add-to-cart event fired from client side to plausible then all required schema attribute must be present and if not then it will go into the logs. 

Q2. and can we create , and update the events through API or it is UI based only?

Q2. Can we change the default database from postgres Database to Mysql database ?

Thanks

**Comment by ruslandoga** [comment]
👋

> Q2. Can we change the default database from postgres Database to Mysql database ?

Right now the answer is "no". For the future it's "highly unlikely", but if you are willing to fork the repo, you might be able to do it as the [Ecto library](https://github.com/elixir-ecto/ecto) which is used to interact with Postgres also has a MySQL adapter.
